selene/src/roblox/collect_std.rs

```diff
@@ -0,0 +1,129 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use chrono::TimeZone;
+use color_eyre::eyre::Context;
+use selene_lib::{standard_library::StandardLibrary, CheckerConfig, RobloxStdSource};
+
+use super::RobloxGenerator;
+
+// Someday, this can be a global config
+const HOUR_CACHE: i64 = 6;
+
+pub fn collect_roblox_standard_library<V>(
+    config: &CheckerConfig<V>,
+    current_directory: &Path,
+) -> color_eyre::Result<StandardLibrary> {
+    let (cached_library, output_directory, output_location) = match config.roblox_std_source {
+        RobloxStdSource::Floating => {
+            let floating_file_directory = floating_file_directory()?;
+            let floating_file = floating_file_directory.join("roblox.yml");
+
+            if floating_file.exists() {
+                let mut library: StandardLibrary = serde_yaml::from_reader(
+                    fs::File::open(&floating_file)
```

Comment:
You should wrap this in a `BufReader`, which can result in major performance improvements for readers!